datafusion/sql/src/unparser/expr.rs

```diff
@@ -2167,6 +2174,39 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_cast_value_to_binary_expr() {
+        let tests = [
+            (
+                Expr::Cast(Cast {
+                    expr: Box::new(Expr::Literal(ScalarValue::Utf8(Some(
+                        "blah".to_string(),
+                    )))),
+                    data_type: DataType::Binary,
+                }),
```

Comment:
I think you can write this more concisely with `lit`

```suggestion
                Expr::Cast(Cast {
                    expr: Box::new(lit("blah"))
                    data_type: DataType::Binary,
                }),
```




You could also try to use https://docs.rs/datafusion/latest/datafusion/logical_expr/enum.Expr.html#method.cast_to  but that requires a valid schema